pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/ComputeStep.java

```diff
@@ -288,4 +303,41 @@ private Schema getAvroSchema(ComputeFieldType type) {
           }
         });
   }
+
+  private org.apache.pulsar.client.api.Schema getPrimitiveSchema(ComputeFieldType type) {
+    org.apache.pulsar.client.api.Schema schema;
+    switch (type) {
+      case STRING:
+        schema = org.apache.pulsar.client.api.Schema.STRING;
+        break;
+      case INT32:
+        schema = org.apache.pulsar.client.api.Schema.INT32;
+        break;
+      case INT64:
+        schema = org.apache.pulsar.client.api.Schema.INT64;
+        break;
+      case FLOAT:
+        schema = org.apache.pulsar.client.api.Schema.FLOAT;
+        break;
+      case DOUBLE:
+        schema = org.apache.pulsar.client.api.Schema.DOUBLE;
+        break;
+      case BOOLEAN:
+        schema = org.apache.pulsar.client.api.Schema.BOOL;
+        break;
+      case DATE:
+        schema = org.apache.pulsar.client.api.Schema.DATE;
+        break;
+      case TIME:
+        schema = org.apache.pulsar.client.api.Schema.TIME;
+        break;
+      case DATETIME:
```

Comment:
I think we should have the same names.
So `ComputeFieldType.TIMESTAMP` -> `Schema.TIMESTAMP`.
And support the other primitive types `INSTANT`, `LOCAL_DATE_TIME`, ...

Reply:
ComputeFieldType doesn't have a 1TIMESTAMP1. The convention used so far (up for feedback, review) is `TIME` `DATE`, `DATETIME`.  These are meant as logical abstraction on top of the Avro schema, in Avro, the underlying java client will resolve to INT/LONG, but in Pulsar API, the schema types are directly coupled with the Java types and hence the confusion. In principle the user can represent any RFC3339 today with the available compute types, but lets have a discussion about what makes sense for the primitive types use cases...

If we don't go with 1:1 mapping for primitive type, we can still add missing types like `BYTE` but `BYTEBUFFER` for example won't be useful.